doozer/doozerlib/cli/fbc.py

```diff
@@ -425,6 +470,175 @@ async def run(self):
+
+        # Check for existing FBC builds if force is not set
+        if not self.force:
+            existing_builds = []
+            for dgk, bundle_build in dgk_bundle_builds.items():
+                operator_meta = runtime.image_map[dgk]
+                existing_fbc_build = await self._check_existing_fbc_build(operator_meta, bundle_build)
+                if existing_fbc_build:
+                    existing_builds.append((dgk, existing_fbc_build))
+
+            if existing_builds:
+                self._logger.info("Found existing FBC builds, skipping rebase and build:")
+                for dgk, existing_build in existing_builds:
+                    self._logger.info(f"  {dgk}: {existing_build.nvr}")
+                self._logger.info("Use --force to override this check and rebuild")
+                return
+
+        # Set up rebase and build components once
+        rebaser = KonfluxFbcRebaser(
+            base_dir=Path(runtime.working_dir, constants.WORKING_SUBDIR_KONFLUX_FBC_SOURCES),
+            group=runtime.group,
+            assembly=assembly,
+            version=self.version,
+            release=self.release,
+            commit_message=self.commit_message,
+            push=not self.dry_run,
+            fbc_repo=self.fbc_repo,
+            upcycle=runtime.upcycle,
+            record_logger=runtime.record_logger,
+        )
+
+        builder = KonfluxFbcBuilder(
+            base_dir=Path(runtime.working_dir, constants.WORKING_SUBDIR_KONFLUX_FBC_SOURCES),
+            group=runtime.group,
+            assembly=runtime.assembly,
+            db=self._fbc_db,
+            fbc_repo=self.fbc_repo,
+            konflux_kubeconfig=self.konflux_kubeconfig,
+            konflux_context=self.konflux_context,
+            konflux_namespace=self.konflux_namespace,
+            image_repo=self.image_repo,
+            skip_checks=self.skip_checks,
+            pipelinerun_template_url=self.plr_template,
+            dry_run=self.dry_run,
+            record_logger=runtime.record_logger,
+        )
+
+        async def _rebase_and_build(operator_meta: ImageMetadata, bundle_build: KonfluxBundleBuildRecord):
+            self._logger.info(f"Rebasing fbc for {operator_meta.name}...")
+            await rebaser.rebase(operator_meta, bundle_build, self.version, self.release)
+            self._logger.info(f"Building fbc for {operator_meta.name}...")
+            await builder.build(operator_meta)
+
+        tasks = [
+            _rebase_and_build(self.runtime.image_map[dgk], bundle_build)
+            for dgk, bundle_build in dgk_bundle_builds.items()
+        ]
+        results = await asyncio.gather(*tasks, return_exceptions=True)
+
+        # Check for failures
+        failed_operators = []
+        for result in results:
+            if isinstance(result, Exception):
+                failed_operators.append("unknown")
+            elif result is not None:
```

Comment:
`_rebase_and_build` doesn't return anything.

Guess you might need `for dgk, result in zip(dgks, results):` to map dgk and result.

Reply:
modified to return nvr now